### DISCUSSION TITLE
FileParserUtils::toInt will fail when input = "   "

### DISCUSSION THREAD

**Discussion by Xu-Yanbo** [rdkit/rdkit#5771] [Development]
Hi,

in rdkit version: 2022_09_1
int FileParserUtils::toInt(const std::string_view input, bool acceptSpaces) will fail when input = "   ",
some molfile often replace "  0" with "   ", so it will make toInt failed.  
e.g.
```



 10 10  0  0  0  0  0  0  0  0999 V2000
    1.4438   -1.0717    0.0000 C   0  0  0  0  0  0  0  0  0  1  0  0
    1.0313   -0.3573    0.0000 C   0  0  0  0  0  0  0  0  0  2  0  0
    1.4438    0.3572    0.0000 N   0  0  0  0  0  0  0  0  0  3  0  0
    0.2063   -0.3573    0.0000 C   0  0  0  0  0  0  0  0  0  4  0  0
   -0.2063   -1.0717    0.0000 C   0  0  0  0  0  0  0  0  0  5  0  0
   -1.0313   -1.0717    0.0000 C   0  0  0  0  0  0  0  0  0  6  0  0
   -1.4438   -0.3573    0.0000 C   0  0  0  0  0  0  0  0  0  7  0  0
   -1.0313    0.3572    0.0000 C   0  0  0  0  0  0  0  0  0  8  0  0
   -1.4438    1.0717    0.0000 Br  0  0  0  0  0  0  0  0  0  9  0  0
   -0.2063    0.3572    0.0000 N   0  0  0  0  0  0  0  0  0 10  0  0
  2  1  1  1        0
  2  3  1  0        0
  2  4  1  0        0
  4  5  4  0        0
  5  6  4  0        0
  6  7  4  0        0
  7  8  4  0        0
  8  9  1  0        0
  8 10  4  0        0
 10  4  4  0        0
M  END
```
![molfile](https://user-images.githubusercontent.com/83994219/202359810-12814ffe-a6f3-4dbc-ad57-23e49109a664.JPG)

I think there is the same question for other functions like toUnsigned.
Thanks!

**Comment by greglandrum** [comment]
@Xu-Yanbo : I'm not able to reproduce this. The problem may have been that your original mol block was missing a blank line at the top. I fixed this above, and what is there parses without complaint.
If you have a file where you observe the problem, please attach it (as `.txt`) to the question.

**Reply by Xu-Yanbo** [reply]
@greglandrum, Thanks for reply!

attach eg.txt will reproduce this problem,  it cause call FileParserUtils::toInt("   ",true);   //the first parameter 'input' is a string of three space

I think the complaint will raise from these code:

'input' is all space string, but  while (*txt == ' ')  don't test when the string end,  'txt' will beyond string length, and sz will be -1. (sz type is unsigned ,so it real be 4294967295).

I think while (*txt == ' '  && sz > 0) will be ok.

https://github.com/rdkit/rdkit/blob/6d10f90a8bd6c97b90f2a3b177d41fd3e6e50fc1/Code/GraphMol/FileParsers/MolFileParser.cpp#L77-L85

[eg.txt](https://github.com/rdkit/rdkit/files/10039073/eg.txt)

**Comment by greglandrum** [comment]
I cannot reproduce this problem. That file reads without problems:
```
In [5]: m = Chem.MolFromMolFile('/home/glandrum/Downloads/eg.txt')

In [6]: m
Out[6]: <rdkit.Chem.rdchem.Mol at 0x7fa208e035e0>
```

**Reply by Xu-Yanbo** [reply]
I use MSVC 2017, maybe std::from_chars in MSVC check more strictly.
if add this code to print sz value :   _**std::cout << sz << std::endl;**_ 
before the follow line , use eg.txt  to check sz value, you will find what happend.
sometimes sz is 4294967294 ,  too big to wrong.

https://github.com/rdkit/rdkit/blob/6d10f90a8bd6c97b90f2a3b177d41fd3e6e50fc1/Code/GraphMol/FileParsers/MolFileParser.cpp#L84-L86
